firedrake/checkpointing.py

```diff
@@ -603,9 +616,10 @@ def save_mesh(self, mesh):
             if mesh.name not in self.require_group(path):
                 path = self._path_to_mesh(tmesh.name, mesh.name)
                 self.require_group(path)
-                # Firedrake coodinates are saved here, but never loaded at the moment.
-                # We load plex coordinates instead.
-                mesh.init()
+                # Firedrake coodinates are saved here, but only loaded for periodic meshes.
+                # Otherwise, we load plex coordinates.
+                if mesh.periodic:
+                    self.set_attr(path, PREFIX_PERIODIC, True)
```

Comment:
This seems like the wrong test. What about another mesh with non-plex coordinates: e.g. high order?

Reply:
I see. We can use `mesh._made_from_coordinates` instead and it's more robust.